File: okhttputil/src/main/java/com/ashlikun/okhttputils/http/cache/BaseCachePolicy.java
package com.ashlikun.okhttputils.http.cache;

import com.ashlikun.okhttputils.http.request.HttpRequest;

/**
 * 作者　　: 李坤
 * 创建时间: 2019/4/4　17:47
 * 邮箱　　：496546144@qq.com
 * <p>
 * 功能介绍：
 */
public abstract class BaseCachePolicy implements CachePolicy {
    protected HttpRequest request;
    protected CacheMode cacheMode;
    protected long cacheTime;

    public BaseCachePolicy(HttpRequest request) {
        this.request = request;
    }

    @Override
    public void setRequest(HttpRequest request) {
        this.request = request;
    }

    @Override
    public void setCacheMode(CacheMode cacheMode) {
        this.cacheMode = cacheMode;
    }

    @Override
    public void setCacheTime(long time) {
        cacheTime = time;
    }

    @Override
    public CacheMode getCacheMode() {
        return cacheMode;
    }


    @Override
    public long getCacheTime() {
        return cacheTime;
    }

    @Override
    public CacheEntity getCache() {
        if (request == null) {
            return null;
        }
        CacheEntity cacheEntity = CacheManage.queryById(CacheManage.getCacheKey(request));
        //检查过期
        if (cacheEntity != null && cacheTime != CacheEntity.CACHE_NEVER_EXPIRE && cacheTime > 0) {
            if (System.currentTimeMillis() - cacheEntity.cacheTime > cacheTime) {
                //过期的
                cacheEntity = null;
            }
        }
        return cacheEntity;
    }
}

File: okhttputil/src/main/java/com/ashlikun/okhttputils/http/cache/BaseCachePolicy.kt
package com.ashlikun.okhttputils.http.cache

import com.ashlikun.okhttputils.http.request.HttpRequest

/**
 * @author　　: 李坤
 * 创建时间: 2021.12.16 10:04
 * 邮箱　　：496546144@qq.com
 *
 * 功能介绍：缓存的基础类
 */

abstract class BaseCachePolicy(override var request: HttpRequest) : CachePolicy {
    override var cacheMode: CacheMode = CacheMode.NO_CACHE
    override var cacheTime: Long = 0
    override val cache: CacheEntity?
        get() {
            var cacheEntity = CacheManage.queryById(CacheManage.getCacheKey(request))
            //检查过期
            if (cacheEntity != null && cacheTime != CacheEntity.CACHE_NEVER_EXPIRE && cacheTime > 0) {
                if (System.currentTimeMillis() - cacheEntity.cacheTime > cacheTime) {
                    //过期的
                    cacheEntity = null
                }
            }
            return cacheEntity
        }

}